### PR TITLE
Improvements to VideoFrame and publish_track APIs

### PR DESCRIPTION
**Pull request by davidzhao**
Using defaults instead of requiring users to pass in these parameters.

This *is* a breaking change. We'll need to release 0.8.0 afterwards.

### CODE REVIEW

File: livekit-rtc/livekit/rtc/video_frame.py [hunk 1]
```diff
@@ -26,10 +26,18 @@
 class VideoFrame:
     def __init__(
         self,
-        timestamp_us: int,
-        rotation: VideoRotation.ValueType,
         buffer: "VideoFrameBuffer",
+        *,
```
Comment (theomonnom) [inline livekit-rtc/livekit/rtc/video_frame.py:30]: This is a breaking change

**Review by theomonnom — APPROVED** [review]
We could avoid the breaking change by removing the *. Tho still nit
otherwise lgtm

**Comment by davidzhao** [comment]
> We could avoid the breaking change by removing the *. Tho nit otherwise lgtm, ty for the docstr

the order of args is still different though.. so I think it'd break either way? Just thinking that it'll only get harder to modify the API.. so might as well do it now 😛

**Comment by theomonnom** [comment]
Oh right, didn't see it

**Pull request merged**